lib/go/test/allday_test.go

```diff
@@ -538,3 +550,116 @@ func testMintMomentNFT(
 		assert.Equal(t, previousSupply, newSupply)
 	}
 }
+
+// ------------------------------------------------------------
+// MomentNFT MetadataViews
+// ------------------------------------------------------------
+func TestMomentNFTMetadataViews(t *testing.T) {
+	b := newEmulator()
+	contracts := AllDayDeployContracts(t, b)
+	userAddress, userSigner := createAccount(t, b)
+	setupAllDay(t, b, userAddress, userSigner, contracts)
+	createTestEditions(t, b, contracts)
+	mintMomentNFT(t, b, contracts, userAddress /*editionID*/, 1 /*shouldRevert*/, false)
+
+	t.Run("Should be able to get moment's metadata", func(t *testing.T) {
+		result := getMomentNFTMetadata(t, b, contracts, userAddress, 1, false)
+
+		//Validate Display
+		displayView := result[0]
+		assert.Equal(t, "Apple Alpha Interception", displayView.Fields[0].ToGoValue())
+		assert.Equal(t, "Series One Set One moment with serial number 1", displayView.Fields[1].ToGoValue())
+		assert.Equal(t, "https://assets.nflallday.com/editions/1/media/image?format=jpeg&width=256",
+			displayView.Fields[2].(cadence.Struct).Fields[0].ToGoValue())
+
+		//Validate Editions
+		editionsView := result[1]
+		edition := editionsView.Fields[0].(cadence.Array).Values[0].(cadence.Struct)
+		assert.Equal(t, "Set One: #1", edition.Fields[0].ToGoValue())
+		assert.Equal(t, uint64(1), edition.Fields[1].ToGoValue())
+		assert.Equal(t, uint64(2), edition.Fields[2].ToGoValue())
+
+		// Validate External URL
+		externalURLView := result[2]
+		assert.Equal(t, "https://nflallday.com/moments/1", externalURLView.Fields[0].ToGoValue())
+
+		//Validate Medias
+		mediasView := result[3]
+		editions := mediasView.Fields[0].(cadence.Array)
+		assert.Equal(t, "https://assets.nflallday.com/editions/1/media/image?format=jpeg&width=512",
+			editions.Values[0].(cadence.Struct).Fields[0].(cadence.Struct).Fields[0].ToGoValue())
```

Comment:
nice tests! You could have a helper func that takes in the index and `editions` and returns the result of 
```
editions.Values[0].(cadence.Struct).Fields[0].(cadence.Struct).Fields[0].ToGoValue())
```

that you could use in the assert.Equal just to make the tests a little less verbose